File: sample/UseLocalLibrary/GNAdSampleNativeAd/app/src/main/java/jp/co/geniee/samples/gnadsamplenativead/GNQueue.java
package jp.co.geniee.samples.gnadsamplenativead;

import java.util.ArrayList;

public class GNQueue {
	int maxSize;
	ArrayList<Object> queue;
	
	public GNQueue(int maxSize) {
		super();
		this.maxSize = maxSize;
		this.queue = new ArrayList<Object>();
	}
	
	public Object dequeue() {
		Object obj;
		if (queue.size() == 0) return null;
		obj = queue.get(0);
		if (obj != null) {
			queue.remove(0);
		}
		return obj;
	}
	
	public void enqueue(Object obj) {
		if (obj == null) return;
		if (queue.size() >= maxSize) {
			queue.remove(0);
		}
		queue.add(obj);
	}
	
	public int size() {
		return queue.size();
	}

}

File: sample/kotlinSample/src/main/java/jp/co/geniee/samples/gnad/common/GNQueue.kt
package jp.co.geniee.samples.gnad.common

import java.util.ArrayList

class GNQueue(private var maxSize: Int) {

    private var queue: ArrayList<Any> = ArrayList()

    fun dequeue(): Any? {
        val obj: Any?
        if (queue.size == 0) return null
        obj = queue[0]
        queue.removeAt(0)

        return obj
    }

    fun enqueue(obj: Any?) {
        if (obj == null) return
        if (queue.size >= maxSize) {
            queue.removeAt(0)
        }
        queue.add(obj)
    }

    fun size(): Int {
        return queue.size
    }

}
